gax/src/test/java/com/google/api/gax/batching/v2/BatcherImplTest.java

```diff
@@ -0,0 +1,214 @@
+    }
+    assertThat(actualError).isInstanceOf(IllegalStateException.class);
+    assertThat(actualError.getMessage()).matches("Cannot add elements on a closed batcher.");
+  }
+
+  /** Verifies unaryCallable is being called with a batch. */
+  @Test
+  public void testResultsAfterRPCSucceed() throws Exception {
+    underTest = createNewBatcherBuilder().setUnaryCallable(mockUnaryCallable).build();
+    when(mockUnaryCallable.futureCall(any(LabeledIntList.class)))
+        .thenReturn(ApiFutures.immediateFuture(Arrays.asList(16, 25)));
+
+    Future<Integer> result = underTest.add(4);
+    Future<Integer> anotherResult = underTest.add(5);
+    underTest.flush();
+
+    assertThat(result.isDone()).isTrue();
+    assertThat(result.get()).isEqualTo(16);
+    assertThat(anotherResult.get()).isEqualTo(25);
+    verify(mockUnaryCallable, times(1)).futureCall(any(LabeledIntList.class));
+  }
+
+  /** Verifies exception occurred at RPC is propagated to element results */
+  @Test
+  public void testResultFailureAfterRPCFailure() throws Exception {
+    underTest = createNewBatcherBuilder().setUnaryCallable(mockUnaryCallable).build();
+    final Exception fakeError = new RuntimeException();
+
+    when(mockUnaryCallable.futureCall(any(LabeledIntList.class)))
+        .thenReturn(ApiFutures.<List<Integer>>immediateFailedFuture(fakeError));
+
+    Future<Integer> failedResult = underTest.add(5);
+    underTest.flush();
+    assertThat(failedResult.isDone()).isTrue();
+    Throwable actualError = null;
+    try {
+      failedResult.get();
+    } catch (InterruptedException | ExecutionException ex) {
+      actualError = ex;
+    }
+
+    assertThat(actualError.getCause()).isSameAs(fakeError);
+    verify(mockUnaryCallable, times(1)).futureCall(any(LabeledIntList.class));
+  }
+
+  /** Tests results are resolves when {@link BatchingDescriptor#splitResponse} throws exception. */
+  @Test
+  public void testExceptionDescriptorResultHandling() throws InterruptedException {
+    underTest = createNewBatcherBuilder().setBatchingDescriptor(mockDescriptor).build();
+
+    final RuntimeException fakeError = new RuntimeException("internal exception");
+    when(mockDescriptor.newRequestBuilder(any(LabeledIntList.class)))
+        .thenReturn(SQUARER_BATCHING_DESC_V2.newRequestBuilder(labeledIntList));
+    doThrow(fakeError)
+        .when(mockDescriptor)
+        .splitResponse(Mockito.<Integer>anyList(), Mockito.<SettableApiFuture<Integer>>anyList());
+
+    Future<Integer> result = underTest.add(2);
+    underTest.flush();
+    Throwable actualError = null;
+    try {
+      result.get();
+    } catch (ExecutionException ex) {
+      actualError = ex;
+    }
+
+    assertThat(actualError.getCause()).isSameAs(fakeError);
+    verify(mockDescriptor)
+        .splitResponse(Mockito.<Integer>anyList(), Mockito.<SettableApiFuture<Integer>>anyList());
+  }
+
+  /** Tests results are resolves when {@link BatchingDescriptor#splitException} throws exception. */
```

Comment:
Resolves test results when {@linkBatchingDescriptor#splitException} throws exception. 

or 

Tests results resolve when ... throws exception. (Editors note: I prefer the first sentence as it starts with an action verb.)